plugins/basic/nexus-repository-maven/src/main/java/org/sonatype/nexus/repository/maven/MavenPath.java

```diff
@@ -335,6 +368,63 @@ public MavenPath signature(final SignatureType signatureType) {
     );
   }
 
+  /**
+   * Returns path pointing to given extension and optionally classifier within this same GAV. Only usable for artifact
+   * paths, those having non-null {@link #getCoordinates()}.
+   */
+  @Nonnull
+  public MavenPath locate(final String extension, @Nullable final String classifier) {
+    checkNotNull(extension);
+    checkArgument(coordinates != null, "Only artifact paths may locate: %s", this);
+
+    MavenPath origin = main();
+    Coordinates coordinates = new Coordinates(
+        origin.coordinates.isSnapshot(),
+        origin.coordinates.getGroupId(),
+        origin.coordinates.getArtifactId(),
+        origin.coordinates.getVersion(),
+        origin.coordinates.getTimestamp(),
+        origin.coordinates.getBuildNumber(),
+        origin.coordinates.getBaseVersion(),
+        classifier,
+        extension,
+        null
+    );
+    // strip ".ext"
+    String newPath = origin.path
+        .substring(0, origin.path.length() - (origin.getCoordinates().getExtension().length() + 1));
+    if (origin.coordinates.classifier != null) {
```

Comment:
minor: inconsistent use of method access vs field access in this code: when trimming the extension it uses `origin.getCoordinates().getExtension()` but when trimming the classifier it uses `origin.coordinates.classifier`